### PR TITLE
Address some cppcheck warnings

### PR DESCRIPTION
**Pull request by sshedi**

### CODE REVIEW

File: client/config.c [hunk 1]
```diff
@@ -119,7 +119,6 @@ TDNFReadConfig(
 
     /* defaults */
     pConf->nGPGCheck = 0;
-    pConf->nInstallOnlyLimit = 1;
```
Comment (oliverkurth) [inline client/config.c:122]: Why are you removing this?

File: llconf/modules.c [hunk 2]
```diff
@@ -237,12 +236,12 @@ int cnfmodule_unparse(struct cnfmodule *cm, FILE* fout,
 int cnfmodule_unparse_file(struct cnfmodule *cm, const char *fname,
 			   struct cnfnode *cn_root)
 {
-	FILE *fout;
 	int ret = -1;
 
 	if(fname == NULL)
 		fname = cm->default_file;
 	if(fname){
+	    FILE *fout;
```
Comment (oliverkurth) [inline llconf/modules.c]: this looks misaligned

File: llconf/nodes.c [hunk 3]
```diff
@@ -377,7 +377,8 @@ char *cnfnode_path(struct cnfnode *cn)
 	*q0 = 0;
 
 	for(cn_parent = cn; cn_parent; cn_parent = cn_parent->parent){
-		char *p, *q;
+		const char *p;
+        char *q;
```
Comment (oliverkurth) [inline llconf/nodes.c:381]: misaligned

File: plugins/metalink/utils.c [hunk 4]
```diff
@@ -471,8 +459,12 @@ TDNFXmlParseData(
...
     if(!elementInfo || !elementInfo->ml_ctx || IsNullOrEmptyString(elementInfo->filename) || (elementInfo->dwError != 0))
     {
-        elementInfo->dwError = ERROR_TDNF_INVALID_PARAMETER;
-        BAIL_ON_TDNF_ERROR(elementInfo->dwError);
+        if (elementInfo) {
+            elementInfo->dwError = ERROR_TDNF_INVALID_PARAMETER;
+            BAIL_ON_TDNF_ERROR(elementInfo->dwError);
+        }
+        pr_err("%s:%d elementInfo is NULL\n", __func__, __LINE__);
+        return;
```
Comment (oliverkurth) [inline plugins/metalink/utils.c]: Good catching the `NULL` pointer check, but I think there should be a `BAIL_ON_TDNF_ERROR` here.

File: solv/tdnfquery.c [hunk 5]
```diff
@@ -505,10 +505,7 @@ SolvGenerateCommonJob(
     {
         queue_empty(&queueJob);
         queue_push2(&queueJob, SOLVER_SOLVABLE_ALL, 0);
-        if (pQuery->queueRepoFilter.count)
-        {
-            selection_filter(pPool, &queueJob, &pQuery->queueRepoFilter);
-        }
+        selection_filter(pPool, &queueJob, &pQuery->queueRepoFilter);
```
Comment (oliverkurth) [inline solv/tdnfquery.c:508]: Why are you removing the `if (...` here?

**Review by oliverkurth — CHANGES_REQUESTED** [review]
Just a few minor comments, but you need to fix the CI failures.

File: client/config.c (hunk 1 above)
Comment (sshedi) [reply client/config.c:122]: Check line 126, this is overwritten

File: solv/tdnfquery.c (hunk 5 above)
Comment (sshedi) [reply solv/tdnfquery.c:508]: We enter inside this block based on same condition, so it's redundant.

File: llconf/nodes.c (hunk 3 above)
Comment (sshedi) [reply llconf/nodes.c:381]: I changed tabs to spaces in this file. Hope that's okay.

**Comment by oliverkurth** [comment]
Looks good, thank you!

**Pull request merged**